pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -355,6 +374,35 @@ type VolumeToMount struct {
 	DesiredSizeLimit *resource.Quantity
 }
 
+// DeviceMountState represents device mount state in a global path.
+type DeviceMountState string
+
+const (
+	// DeviceGloballyMounted means device has been globally mounted successfully
+	DeviceGloballyMounted DeviceMountState = "DeviceGloballyMounted"
+
+	// DeviceMountUncertain means device may not be mounted but a mount operation may be
+	// in-progress which can cause device mount to succeed.
+	DeviceMountUncertain DeviceMountState = "DeviceMountUncertain"
+
+	// DeviceNotMounted means device has not been mounted globally.
+	DeviceNotMounted DeviceMountState = "DeviceNotMounted"
+)
+
+// VolumeMountState represents volume mount state in a path local to the pod.
+type VolumeMountState string
+
+const (
+	// VolumeMounted means volume has been mounted in pod's local path
+	VolumeMounted VolumeMountState = "VolumeMounted"
+
+	// VolumeMountUncertain means volume may or may not be mounted in pods' local path
+	VolumeMountUncertain VolumeMountState = "VolumeMountUncertain"
+
+	// VolumeNotMounted means volume has not been mounted in pod's local path
+	VolumeNotMounted VolumeMountState = "VolumeNotMounted"
```

Comment:
do we really need this state?
Normally we can just use a boolean value to indicate the volume is already mounted if it is true. If it is false, that means we are not sure it is mounted or not, but still add it into the state and reconciler will update the state after it successfully mounted.

Reply:
Never mind. Actually I see the point of having three states.

Reply:
We can manage with boolean value of mount state but in a lot of places we typically have to carry a comment to explain a boolean value (/*MountUncertain*/) and stuff like that. I thought it might be best to explicit and let the field document what it is and that way - it will be more clear. What do you think?

Reply:
Another thing I am thinking is - if mount failed with timeout first time and on second attempt mount failed with non-timeout(final) error then the volume should be moved to `VolumeNotMounted` state. I will update the code to do that.